stage3/buf.C

```diff
@@ -68,7 +68,37 @@ const Status BufMgr::unPinPage(File *file, const int PageNo, const bool dirty) {
 
 }
 
-const Status BufMgr::allocPage(File *file, int &pageNo, Page *&page) {}
+const Status BufMgr::allocPage(File *file, int &pageNo, Page *&page) {
+  if (file == nullptr)
+    return UNIXERR;
+  auto status = file->allocatePage(pageNo);
```

Comment:
Have you tried to compile the project on a CS machine? Not sure if C++11 is enabled by default. Since we cannot change the `Makefile`, we have to make sure that our code is valid under the default C++ language standard used by their compiler.